test/e2e/disruption.go

```diff
@@ -66,73 +68,98 @@ var _ = framework.KubeDescribe("DisruptionController", func() {
 
 	})
 
-	It("should allow an eviction when there is no PDB", func() {
-		createPodsOrDie(cs, ns, 1)
-
-		pod, err := cs.Pods(ns).Get("pod-0")
-		Expect(err).NotTo(HaveOccurred())
-
-		e := &policy.Eviction{
-			ObjectMeta: api.ObjectMeta{
-				Name:      pod.Name,
-				Namespace: ns,
-			},
-		}
-
-		err = cs.Pods(ns).Evict(e)
-		Expect(err).NotTo(HaveOccurred())
-	})
-
-	It("should not allow an eviction when too few pods", func() {
-		createPodDisruptionBudgetOrDie(cs, ns, intstr.FromInt(2))
-
-		createPodsOrDie(cs, ns, 1)
-
-		pod, err := cs.Pods(ns).Get("pod-0")
-		Expect(err).NotTo(HaveOccurred())
-
-		e := &policy.Eviction{
-			ObjectMeta: api.ObjectMeta{
-				Name:      pod.Name,
-				Namespace: ns,
-			},
+	evictionCases := []struct {
+		description    string
+		minAvailable   intstr.IntOrString
+		podCount       int
+		replicaSetSize int32
+		shouldDeny     bool
+	}{
+		{
+			description:  "no PDB",
+			minAvailable: intstr.FromString(""),
+			podCount:     1,
+			shouldDeny:   false,
+		}, {
+			description:  "too few pods, absolute",
+			minAvailable: intstr.FromInt(2),
+			podCount:     1,
+			shouldDeny:   true,
+		}, {
+			description:  "enough pods, absolute",
+			minAvailable: intstr.FromInt(2),
+			podCount:     2,
+			shouldDeny:   false,
+		}, {
+			description:    "enough pods, replicaSet, percentage",
```

Comment:
I had been trying to figure out how to force a controller to be at <100%.  The best I can come up with is to put an impossible node selector on it to make 0% of them be ready.  WDYT?